crates/engine/tree/src/persistence.rs

```diff
@@ -194,27 +203,29 @@ pub enum PersistenceAction<N: NodePrimitives = EthPrimitives> {
 #[derive(Debug, Clone)]
 pub struct PersistenceHandle<N: NodePrimitives = EthPrimitives> {
     /// The channel used to communicate with the persistence service
-    sender: Sender<PersistenceAction>,
-    _marker: std::marker::PhantomData<N>,
+    sender: Sender<PersistenceAction<N>>,
 }
 
 impl<T: NodePrimitives> PersistenceHandle<T> {
     /// Create a new [`PersistenceHandle`] from a [`Sender<PersistenceAction>`].
-    pub const fn new(sender: Sender<PersistenceAction>) -> Self {
-        Self { sender, _marker: std::marker::PhantomData }
+    pub const fn new(sender: Sender<PersistenceAction<T>>) -> Self {
+        Self { sender }
     }
 
     /// Create a new [`PersistenceHandle`], and spawn the persistence service.
-    pub fn spawn_service<N: PersistenceNodeTypes>(
+    pub fn spawn_service<N>(
         provider_factory: ProviderFactory<N>,
         pruner: PrunerWithFactory<ProviderFactory<N>>,
         sync_metrics_tx: MetricEventsSender,
-    ) -> Self {
+    ) -> PersistenceHandle<N::Primitives>
+    where
+        N: PersistenceNodeTypes,
+    {
         // create the initial channels
         let (db_service_tx, db_service_rx) = std::sync::mpsc::channel();
 
         // construct persistence handle
-        let persistence_handle = Self::new(db_service_tx);
+        let persistence_handle = PersistenceHandle::<N::Primitives>::new(db_service_tx);
```

Comment:
If we do just `PersistenceHandle::new` here, does it still compile?

Reply:
yes, updated.